src/widgets/powered-by/powered-by.js

```diff
@@ -0,0 +1,94 @@
+import React, { render, unmountComponentAtNode } from 'preact-compat';
+import cx from 'classnames';
+
+import PoweredBy from '../../components/PoweredBy/PoweredBy';
+import connectPoweredBy from '../../connectors/powered-by/connectPoweredBy.js';
+import { getContainerNode } from '../../lib/utils.js';
+import { component } from '../../lib/suit';
+
+const suit = component('PoweredBy');
+
+const renderer = ({ containerNode, cssClasses }) => (
+  { widgetParams },
+  isFirstRendering
+) => {
+  if (isFirstRendering) {
+    const { url, theme } = widgetParams;
+
+    render(
+      <PoweredBy cssClasses={cssClasses} url={url} theme={theme} />,
+      containerNode
+    );
+
+    return;
+  }
+};
+
+const usage = `Usage:
+poweredBy({
+  container,
+  [ theme = 'light' ],
+})`;
+
+/**
+ * @typedef {Object} PoweredByWidgetOptions
+ * @property {string|HTMLElement} container Place where to insert the widget in your webpage.
+ * @property {string} [theme] The theme of the logo ("light" or "dark").
+ * @property {PoweredByWidgetCssClasses} [cssClasses] CSS classes to add.
+ */
+
+/**
+ * The `poweredBy` widget is used to display useful insights about the current results.
+ * @type {WidgetFactory}
+ * @devNovel PoweredBy
+ * @category metadata
+ * @param {PoweredByWidgetOptions} $0 PoweredBy widget options. Some keys are mandatory: `container`,
+ * @return {Widget} A new poweredBy widget instance
+ * @example
+ * search.addWidget(
+ *   instantsearch.widgets.poweredBy({
+ *     container: '#poweredBy-container'
+ *   })
+ * );
+ */
+export default function poweredBy({
+  container,
+  cssClasses: userCssClasses = {},
+  theme = 'light',
+} = {}) {
+  if (!container) {
+    throw new Error(usage);
+  }
+
+  const containerNode = getContainerNode(container);
+  const url =
```

Comment:
The URL should be in the connector but it might be possible to override it. We can also make the `utm_source` configurable then Vue & Angular can also use the connector to create their PoweredBy component. See [Vue - PoweredBy](https://github.com/algolia/vue-instantsearch/blob/feat/connectors/src/components/PoweredBy.vue)